staging/src/k8s.io/apiextensions-apiserver/pkg/client/informers/externalversions/factory.go

```diff
@@ -120,6 +131,48 @@ func (f *sharedInformerFactory) Start(stopCh <-chan struct{}) {
 	}
 }
 
+// informerStopped removes a stopped informer from the factory's list
+// of informers and started informers.
+func (f *sharedInformerFactory) informerStopped(informerType reflect.Type) {
+	f.lock.Lock()
+	defer f.lock.Unlock()
+	delete(f.startedInformers, informerType)
+	delete(f.informers, informerType)
+}
+
+// StartWithStopOptions initializes all requested informers with the stop options provided, defaulting to
+// the old stop options (only stopping via closure of stopCh).
+// It makes sure remove an informer from the list of informers and started informers when the informer is stopped
+// to prevent a race where InformerFor gives the user back a stopped informer that will never be started again.
+func (f *sharedInformerFactory) StartWithStopOptions(stopCh <-chan struct{}) {
```

Comment:
There are a bunch of factories like this one in various repos in staging (kube-aggregator, sample-controller, sample-apiserver, code-generator). Is this code autogenerated somewhere or do I need to copy/paste in all the various `*/externalversions/factory.go` and factory interfaces?